src/main/java/io/antmedia/datastore/db/MapBasedDataStore.java

```diff
@@ -1055,4 +1131,28 @@ public Broadcast getBroadcastFromMap(String streamId)
 		return null;
 	}
 
+	@Override
+	public boolean blackListToken(Token token) {
+		boolean result = false;
+
+		synchronized (this) {
```

Comment:
I'd revisit the usage of `synchronized` in this class, as there seems to be some overuse and overscoping. Using `this` as a monitor is not very fortunate, as it can be held externally. Here the monitor is obtained too early, only `saveToken` seems to be critical, but as far as I can see `saveToken` itself also takes care of synchronization using `this`, so this might be completely unnecessary.

Reply:
agree